frontend/server/k8s-helper.ts

```diff
@@ -165,11 +166,78 @@ export async function getTensorboardInstance(logdir: string): Promise<string> {
     );
 }
 
+/**
+ * Finds a running Tensorboard instance created via CRD with the given logdir
+ * and returns its version.
+ */
+export async function getTensorboardInstanceVersion(logdir: string): Promise<string> {
+  if (!k8sV1CustomObjectClient) {
+    throw new Error('Cannot access kubernetes Custom Object API');
+  }
+
+  return await k8sV1CustomObjectClient
+    .getNamespacedCustomObject(
+      viewerGroup,
+      viewerVersion,
+      namespace,
+      viewerPlural,
+      getNameOfViewerResource(logdir),
+    )
+    .then(
+      // Viewer CRD pod has tensorboard instance running at port 6006 while
+      // viewer CRD service has tensorboard instance running at port 80. Since
+      // we return service address here (instead of pod address), so use 80.
+      (viewer: any) => {
+        return viewer &&
+          viewer.body &&
+          viewer.body.spec.tensorboardSpec.logDir == logdir &&
+          viewer.body.spec.type == 'tensorboard' &&
+          viewer.body.spec.tensorboardSpec.tensorflowImage
+          ? viewer.body.spec.tensorboardSpec.tensorflowImage.replace('tensorflow/tensorflow:', '')
+          : ''; // default version(also the only version choice in previous kfp)
+      },
+      // No existing custom object with the given name, i.e., no existing
+      // tensorboard instance.
+      (error: any) => '',
+    );
+}
+
+/**
+ * Find a running Tensorboard instance with the given logdir, delete the instance
+ * and returns the deleted podAddress
+ */
+
+export async function deleteTensorboardInstance(logdir: string, tfversion: string): Promise<void> {
+  if (!k8sV1CustomObjectClient) {
+    throw new Error('Cannot access kubernetes Custom Object API');
+  }
+  const currentPod = await getTensorboardInstance(logdir);
```

Comment:
Should getTensorboardInstanceVersion replace getTensorboardInstance, or we want to keep both?

Reply:
I think we can keep both. getTensorboardInstance returns the pod address, while getTensorboardInstanceVersion returns the version. Since separate 2 functions didn't cause obvious latency, keeping both may make their functionality more clear and flexible to future changes.

Reply:
So we decided to keep only `getTensorboardInstance`? SGTM